src/main/java/com/google/finapp/SpannerDao.java

```diff
@@ -16,133 +16,183 @@
 
 import com.google.cloud.ByteArray;
 import com.google.cloud.spanner.*;
-import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import com.google.inject.Inject;
 
 import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Arrays;
 
 final class SpannerDao {
 
-  private final DatabaseClient databaseClient;
+  private final String projectId;
+  private final String databaseId;
+  private final String instanceId;
+  private final String connectionUrl;
 
   @Inject
-  SpannerDao(DatabaseClient databaseClient) {
-    this.databaseClient = databaseClient;
+  SpannerDao(@ArgsModule.SpannerProjectId String spannerProjectId,
```

Comment:
I saw in the PR description that re-using the SpannerDao class is intentional and will change, but why not change it to SpannerDaoJDBC as part of this PR?

Reply:
It will be changed in the next draft of this PR. I left the first draft with the same file name so as to let reviewers see the direct comparison between the Java client and JDBC implementations.